src/Files.App/Services/Windows/WindowsDialogService.cs

```diff
@@ -104,17 +93,7 @@ public unsafe bool Open_FileSaveDialog(nint hWnd, bool pickFoldersOnly, string[]
 			try
 			{
 				using ComPtr<IFileSaveDialog> pDialog = default;
-				var dialogInstanceIid = typeof(FileSaveDialog).GUID;
-				var dialogIid = typeof(IFileSaveDialog).GUID;
-
-				// Get a new instance of the dialog
-				HRESULT hr = PInvoke.CoCreateInstance(
-					&dialogInstanceIid,
-					null,
-					CLSCTX.CLSCTX_INPROC_SERVER,
-					&dialogIid,
-					(void**)pDialog.GetAddressOf())
-				.ThrowOnFailure();
+				HRESULT hr = pDialog.CoCreateInstance<FileSaveDialog>(CLSCTX.CLSCTX_INPROC_SERVER);
```

Comment:
Should we introduce an overload that takes in a CLSID pointer?

Not a blocker though.

Reply:
I think L61 is not AoT-safe? In that case, we should replace instead.

https://github.com/files-community/Files/blob/72c3faf29f4458cd351076a41e3d3936fcadaecf/src/Files.App.CsWin32/ComPtr%601.cs#L58-L63

Reply:
It is safe, as the compiler can reason about what type it should preserve here, but it causes the compiler to root them which can increase binary size (granted, probably not much)

Reply:
Okay then, lets pass the CLSID pointer instead.